processing/src/main/java/io/druid/segment/filter/DimensionPredicateFilter.java

```diff
@@ -37,27 +38,54 @@
 {
   private final String dimension;
   private final Predicate<String> predicate;
+  private final ExtractionFn extractionFn;
 
   public DimensionPredicateFilter(
       String dimension,
-      Predicate<String> predicate
+      Predicate<String> predicate,
+      ExtractionFn extractionFn
   )
   {
     this.dimension = dimension;
     this.predicate = predicate;
+    this.extractionFn = extractionFn;
   }
 
   @Override
   public ImmutableBitmap getBitmapIndex(final BitmapIndexSelector selector)
   {
-    Indexed<String> dimValues = selector.getDimensionValues(dimension);
-    if (dimValues == null || dimValues.size() == 0 || predicate == null) {
+    if (predicate == null) {
       return selector.getBitmapFactory().makeEmptyImmutableBitmap();
     }
+    Indexed<String> dimValues = selector.getDimensionValues(dimension);
+    if (dimValues == null || dimValues.size() == 0) {
```

Comment:
I don't think `dimValues.size() == 0` is correct; a single value doesn't HAVE to be null. This should also check `dimValues.get(0) == null`